InformationScripting/src/parsing/QueryParser.h

```diff
@@ -60,10 +61,10 @@ class INFORMATIONSCRIPTING_API QueryParser
 		Type typeOf(const QString& text);
 		QPair<QStringList, QList<QChar> > split(const QString& text, const QList<QChar>& splitChars);
 
-		Query* parseQuery(const QString& text);
-		QList<Query*> parseList(const QString& text);
-		Query* parseOperator(const QString& text, bool connectInput = false);
-		QList<Query*> parseOperatorPart(const QString& text);
+		Query* parseQuery(Query* parent, const QString& text);
```

Comment:
So I see that we added the parent of each query. I know we talked about this, but somehow I'm not happy with either solution. The core issue is that most queries, don't care about their parents or about the executor, and are standalone units. Now, because one query might need this we make it a requirement for all to know about the parent/executor. Can you think of a way to provide this information only to queries that care about this? Perhaps when creating a query call a virtual setExecutor method which most queries ignore by default, but the ones that care about it, record it? Or something along those lines. What do you think?